playbooks/tasks/portal-mongo-replicaset-check-majority.yml

```diff
@@ -0,0 +1,89 @@
+---
+
+# Check there is still MongoDB majority if we turn off hosts in the current batch
+
+- name: Check mongo container is running
+  community.docker.docker_container_info:
+    name: mongo
+  register: mongo_docker_container_result
+
+# If the mongo container is already stopped we can skip MongoDB replicaset
+# majority check, it doesn't matter that we stop the rest of services.
+- block:
+    # Set mongo db config from .env file (without using LastPass/secrets storage)
+
+    - name: Get mongo user on host_item from .env file
+      ansible.builtin.shell: ". {{ webportal_dir }}/.env && echo $SKYNET_DB_USER"
+      register: skynet_db_user_result
+
+    - name: Get mongo pass on host_item from .env file
+      ansible.builtin.shell: ". {{ webportal_dir }}/.env && echo $SKYNET_DB_PASS"
+      register: skynet_db_pass_result
+
+    - name: Set mongo db config
+      set_fact:
+        mongodb_config:
+          skynet_db_user: "{{ skynet_db_user_result.stdout }}"
+          skynet_db_pass: "{{ skynet_db_pass_result.stdout }}"
+
+    - name: Include checking mongo replicaset majority
+      include_tasks: tasks/portal-role-task-mongo-shell-eval.yml
+      vars:
+        mongodb_shell:
+          eval: |
+            votingMembers = rs.config().members
+              .filter(m => m.priority > 0)
+              .map(m => m.host)
+            onlineVotingMembers = rs.status().members
+              .filter(m => m.stateStr == 'PRIMARY' || m.stateStr == 'SECONDARY')
+              .map(m => m.name)
+              .filter(m => votingMembers.includes(m))
```

Comment:
This should also filter by `health=1`, otherwise we'll be counting crashed nodes as online.

Reply:
Added.